okhttp/src/main/java/okhttp3/internal/Platform.java

```diff
@@ -274,6 +298,58 @@ public Android(Class<?> sslParametersClass, OptionalMethod<Socket> setUseSession
   }
 
   /**
+   * OpenJDK 9+.
+   */
+  private static final class Jdk9Platform extends Platform {
+    private Method setProtocolMethod = null;
+    private Method getProtocolMethod = null;
+
+    public Jdk9Platform(Method setProtocolMethod, Method getProtocolMethod) {
+      this.setProtocolMethod = setProtocolMethod;
+      this.getProtocolMethod = getProtocolMethod;
+    }
+
+    @Override public void configureTlsExtensions(SSLSocket sslSocket, String hostname,
+        List<Protocol> protocols) {
+      if (this.setProtocolMethod != null) {
+        try {
+          SSLParameters sslParameters = sslSocket.getSSLParameters();
+
+          List<String> names = alpnProtocolNames(protocols);
+
+          setProtocolMethod.invoke(sslParameters,
+              new Object[]{names.toArray(new String[names.size()])});
+
+          sslSocket.setSSLParameters(sslParameters);
+        } catch (IllegalAccessException | InvocationTargetException e) {
+          throw new AssertionError();
+        }
+      }
+    }
+
+    @Override public String getSelectedProtocol(SSLSocket socket) {
+      if (this.getProtocolMethod != null) {
```

Comment:
this is always true. There’s no way to get an instance of this class with a null getProtocolMethod